From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Wed, 20 Jun 2018 01:01:03 -0400
Subject: fix Tor test

tor-test.js was assigning a javascript object to
EnigmailGpg.agentPath, rather than just a raw path.

This caused autocrypt-test.js to break, because EnigmailGpg.agentPath
is supposed to just be a simple string.

Fixing tor-test.js, and fixing tor.jsm to do the right thing, makes
the rest of the test suite complete successfully.
---
 package/tests/tor-test.js | 8 ++------
 package/tor.jsm           | 2 +-
 2 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/package/tests/tor-test.js b/package/tests/tor-test.js
index 50029ec..f4a0e56 100644
--- a/package/tests/tor-test.js
+++ b/package/tests/tor-test.js
@@ -56,17 +56,13 @@ test(function evaluateMeetsMinimumCurlSocksVersion() {
 });
 
 test(withEnigmail(function createHelperArgsForTorsocks1(enigmail) {
-  EnigmailGpg.setAgentPath({
-    path: "/usr/bin/gpg2"
-  });
+  EnigmailGpg.setAgentPath("/usr/bin/gpg2");
   const firstSet = createHelperArgs("torsocks", false);
   Assert.deepEqual(firstSet[0], "/usr/bin/gpg2");
 }));
 
 test(function createHelperArgsForTorsocks2() {
-  EnigmailGpg.setAgentPath({
-    path: "/usr/bin/gpg"
-  });
+  EnigmailGpg.setAgentPath("/usr/bin/gpg");
   const args = createHelperArgs("torsocks2", true);
 
   Assert.deepEqual(args[0], "--user");
diff --git a/package/tor.jsm b/package/tor.jsm
index a206887..48da295 100644
--- a/package/tor.jsm
+++ b/package/tor.jsm
@@ -112,7 +112,7 @@ function createHelperArgs(helper, addAuth) {
   if (addAuth) {
     args = ["--user", createRandomCredential(), "--pass", createRandomCredential()];
   }
-  args.push(EnigmailGpg.agentPath.path);
+  args.push(EnigmailGpg.agentPath);
   return args;
 }
 
